### PR TITLE
Basic bank statement

### PR DESCRIPTION
**Pull request by stefan-b-jakobsson**
Implementation of a new BANK statement in BASIC.

The bank set by BANK is used by SYS, POKE and PEEK if the target address is >= $a000

Example (starts GEOS by calling $C000 in ROM bank 3):
```
BANK 3
SYS $C000
```

### CODE REVIEW

File: basic/code17.s [hunk 1]
```diff
@@ -51,24 +51,34 @@ peek	lda poker+1
...
 	txa
 	ldy #0
 	sta (poker),y
 	rts
+pokefr	lda #poker
+	sta stavec
+	txa
+	ldx curbank
+	ldy #0
+	jsr stash
```
Comment (mist64) [inline basic/code17.s:80]: combine to `jsr` & `rts` to `jmp stash`

File: basic/token2.s [hunk 2]
```diff
@@ -81,6 +81,7 @@ reslst2	.byt "MO", 'N' + $80
 	.byt "BLOA", 'D' + $80
 	.byt "BVLOA", 'D' + $80
 	.byt "BVERIF", 'Y' + $80
+	.byt "BAN", 'K'+$80
```
Comment (mist64) [inline basic/token2.s:84]: "space plus space" for consistency

File: basic/init.s [hunk 3]
```diff
@@ -14,6 +14,7 @@ nready	jmp readyx
 init	jsr initv       ;go init vectors
 	jsr initcz      ;go init charget & z-page
 	jsr initms      ;go print initilization messages
+	stz curbank	;set default value for BANK statement
```
Comment (mist64) [inline basic/init.s:17]: spaces for indenting the comments, please

File: cfg/x16.cfginc [hunk 4]
```diff
@@ -33,7 +33,8 @@ KVECTORS: start = $0314, size = $0020; # KERNAL vectors
 /* $0334-$03FF: variables and RAM code (not available for *legacy* GEOS apps) */
 KVAR2:    start = $0334, size = $0057; # KERNAL: screen editor table
 KERNRAM2: start = $038B, size = $003F; # KERNAL: banked IRQ, fetch, stash
-/*        start = $03CA, size = $0002; # reserved for KERNAL growth */
+BVARS2:   start = $03CA, size = $0001; # BASIC
```
Comment (mist64) [inline cfg/x16.cfginc:36]: no need to create a new segment; at this point, it's okay to extend the BVARS segment by 1 byte, moving FPVARS up by 1 byte – we don't try to guarantee binary compatibility with these addresses yet

**Comment by stefan-b-jakobsson** [comment]
I think I fixed all comments from @mist64

**Comment by stefan-b-jakobsson** [comment]
By mistake a change in the GEOS launch code intended for my other PR (fix-gets) had sneaked in here. It was removed with git revert (commit 780b786) above.

**Review by mist64 — APPROVED** [review]

**Comment by irmen** [comment]
Should the LOAD commands also consider the BANK ?

**Comment by mist64** [comment]
> Should the LOAD commands also consider the BANK ?

LOAD and SAVE could benefit from this, yes! Do you want to file an issue? :)

**Pull request merged**